### PR TITLE
F2F-377: Added KMS signing key

### PR DESCRIPTION
**Pull request by AyeshaKhanGDS**
IPV Return requires a KMS signing key

### What changed

- Added the KMS key template
- Associated workflow files

### Issue tracking

- [F2F-377](https://govukverify.atlassian.net/browse/F2F-377)

### Environment variables or secrets

- [ ] Documented in the [README](./blob/main/README.md)
- [ ] Added to deployment repository
- [ ] Added to local startup repository

### Other considerations

- [ ] Update [README](./blob/main/README.md) with any new instructions or tasks

[F2F-377]: https://govukverify.atlassian.net/browse/F2F-377?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: infra-l2-kms/template.yaml [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        Statement:
+          - Effect: Allow
+            Principal:
+              AWS: !Sub 'arn:aws:iam::${AWS::AccountId}:root'
+            Action:
+              - "kms:*"
+            Resource:
+              - "*"
+      KeySpec: RSA_2048
+      KeyUsage: ENCRYPT_DECRYPT
```
Comment (CoralSmith) [inline infra-l2-kms/template.yaml:80]: Oh no! This needs to be a signing key! Can we swap the Usage!
```suggestion
      KeyUsage: SIGN_VERIFY
```

File: infra-l2-kms/template.yaml [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+          Value: VC Signing Key
+        - Key: Environment
+          Value: !Sub ${Environment}
+        - Key: CheckovRulesToSkip
+          Value: CKV_AWS_7
+
+  KMSSigningKeyAlias:
+    Type: 'AWS::KMS::Alias'
+    Properties:
+      AliasName: !Sub "alias/${AWS::StackName}-vc-signing-key"
```
Comment (CoralSmith) [inline infra-l2-kms/template.yaml:94]: This won't be for signing VCs, just for bog standard object signing when making requests to auth for tokens. 

```suggestion
      AliasName: !Sub "alias/${AWS::StackName}-auth-signing-key"
```

File: infra-l2-kms/template.yaml [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+              - "kms:*"
+            Resource:
+              - "*"
+      KeySpec: RSA_2048
+      KeyUsage: ENCRYPT_DECRYPT
+      MultiRegion: false
+      PendingWindowInDays: !Ref KmsPendingWindowInDays
+      Tags:
+        - Key: KeyType
+          Value: VC Signing Key
```
Comment (CoralSmith) [inline infra-l2-kms/template.yaml:85]:
```suggestion
          Value: Signing Key
```

File: infra-l2-kms/template.yaml [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+      - UsePermissionsBoundary
+      - !Ref PermissionsBoundary
+      - !Ref AWS::NoValue
+
+Resources:
+  KMSSigningKey:
+    Type: AWS::KMS::Key
+    #checkov:skip=CKV_AWS_7:Automatic key rotation can only be enabled on symmetric keys.
+    Properties:
+      Description: !Sub "${AWS::StackName} VC signing key"
```
Comment (CoralSmith) [inline infra-l2-kms/template.yaml:66]:
```suggestion
      Description: !Sub "${AWS::StackName} signing key for public_key_jwt method of client assertion"
```

File: infra-l2-kms/template.yaml [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+
+  KMSSigningKeyAlias:
+    Type: 'AWS::KMS::Alias'
+    Properties:
+      AliasName: !Sub "alias/${AWS::StackName}-vc-signing-key"
+      TargetKeyId: !Ref KMSSigningKey
+
+Outputs:
+  SigningKeyArn:
+    Description: "The Arn of the VC signing key"
```
Comment (CoralSmith) [inline infra-l2-kms/template.yaml:99]:
```suggestion
    Description: "The Arn of the public_key_jwt signing key"
```

File: infra-l2-kms/template.yaml [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+    Value: !GetAtt KMSSigningKey.Arn
+    Export:
+      Name: !Sub "${AWS::StackName}-vc-signing-key"
+  SigningKeyId:
+    Description: "The Id of the VC signing key"
+    Value: !Ref KMSSigningKey
+    Export:
+      Name: !Sub "${AWS::StackName}-vc-signing-key-id"
```
Comment (CoralSmith) [inline infra-l2-kms/template.yaml:107]:
```suggestion
      Name: !Sub "${AWS::StackName}-auth-signing-key-id"
```

File: infra-l2-kms/template.yaml [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+  SigningKeyAlias:
+    Description: "The Alias of the VC signing key"
```
Comment (CoralSmith) [inline infra-l2-kms/template.yaml:109]:
```suggestion
    Description: "The Alias of the signing key"
```

**Review by CoralSmith — APPROVED** [review]

**Review by richa-misra — APPROVED** [review]

**Pull request merged**